query/math.go

```diff
@@ -42,13 +42,70 @@ func (v *MathValuer) Value(key string) (interface{}, bool) {
 
 func (v *MathValuer) Call(name string, args []influxql.Expr) (interface{}, bool) {
 	if len(args) == 1 {
+		arg0, ok := v.getFloatValue(args[0])
+		if !ok {
+			return nil, false
+		}
+		
+		var fn func(float64) float64
 		switch name {
+		case "abs":
+			fn = math.Abs
 		case "sin":
-			return v.callTrigFunction(math.Sin, args[0])
+			fn =  math.Sin
```

Comment:
Sounds fine to me. You would likely need to do something like this:

```golang
switch name {
case "sin", "cos", ... all of the other functions that return float ...:
    arg0, ok := asFloat(args[0])
    if !ok {
        return nil, true
    }

    var res float64
    // Need to do another switch to allow inlining.
    switch name {
    case "sin":
        res = math.Sin(arg0))
    default:
        panic("need to panic here")
    }

    if math.IsNaN(res) {
        return nil, true
    }
    return res, true
case "abs":
    // Handled differently because it has different code depending on whether the input
    // is a float or integer.
}
```

It would be a tiny bit more repetitive with some extra typing (plus, you have to do a name switch twice). I don't think that's a big deal though.

I was thinking about it though and you probably don't want `handleNaN`. The main issue is that it would have to return an interface so returning a value type as an `interface{}` usually results in an allocation. That would likely undo any extra optimizations from inlining.

Before we talk more about optimization though, can you implement this the "dumb" way and then we'll make a benchmark? I wouldn't want to worry about any premature optimizations.

As an aside, I just realized the current code is wrong for string and boolean inputs. When it encounters something like `sin(x::string)`, it should return `nil, true`, but it currently returns `nil, false`. You can feel free to fix that. And remember that `NaN` should return `nil, true` too.